app/main.py

```diff
@@ -1,3 +1,58 @@
+from typing import Union
+from functools import total_ordering
+
+
+@total_ordering
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: Union[int, float]) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union["Distance", int, float]) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        return NotImplemented
```

Comment:
In the __add__ method, you're returning NotImplemented if the 'other' argument isn't an instance of Distance, int, or float. This is a good practice, but it might be beneficial to also raise a TypeError with a descriptive message to make debugging easier in case of incorrect usage.